src/Language/PureScript/Interactive/Directive.hs

```diff
@@ -8,62 +8,63 @@ import Prelude
 import Data.Maybe (fromJust)
 import Data.List (isPrefixOf)
 import Data.Tuple (swap)
+import Data.List.NonEmpty (NonEmpty(..))
+import Data.List.NonEmpty qualified as NEL
 
 import Language.PureScript.Interactive.Types (Directive(..))
 
 -- |
 -- A mapping of directives to the different strings that can be used to invoke
 -- them.
 --
-directiveStrings :: [(Directive, [String])]
+directiveStrings :: [(Directive, NonEmpty String)]
 directiveStrings =
-    [ (Help      , ["?", "help"])
-    , (Quit      , ["quit"])
-    , (Reload    , ["reload"])
-    , (Clear     , ["clear"])
-    , (Browse    , ["browse"])
-    , (Type      , ["type"])
-    , (Kind      , ["kind"])
-    , (Show      , ["show"])
-    , (Paste     , ["paste"])
-    , (Complete  , ["complete"])
-    , (Print     , ["print"])
+    [ (Help      , NEL.fromList ["?", "help"])
+    , (Quit      , NEL.singleton "quit")
+    , (Reload    , NEL.singleton "reload")
+    , (Clear     , NEL.singleton "clear")
+    , (Browse    , NEL.singleton "browse")
+    , (Type      , NEL.singleton "type")
+    , (Kind      , NEL.singleton "kind")
+    , (Show      , NEL.singleton "show")
+    , (Paste     , NEL.singleton "paste")
+    , (Complete  , NEL.singleton "complete")
+    , (Print     , NEL.singleton "print")
     ]
 
 -- |
--- Like directiveStrings, but the other way around.
+-- Like `directiveStrings`, but the other way around.
 --
-directiveStrings' :: [(String, Directive)]
-directiveStrings' = concatMap go directiveStrings
-  where
-  go (dir, strs) = map (, dir) strs
+directiveStrings' :: [(NonEmpty String, Directive)]
+directiveStrings' = map (\(d, s) -> (s, d)) directiveStrings
 
 -- |
 -- Returns all possible string representations of a directive.
 --
-stringsFor :: Directive -> [String]
+stringsFor :: Directive -> NonEmpty String
 stringsFor d = fromJust (lookup d directiveStrings)
 
 -- |
 -- Returns the default string representation of a directive.
 --
 stringFor :: Directive -> String
-stringFor = head . stringsFor
+stringFor = NEL.head . stringsFor
 
 -- |
 -- Returns the list of directives which could be expanded from the string
 -- argument, together with the string alias that matched.
 --
-directivesFor' :: String -> [(Directive, String)]
+directivesFor' :: String -> [(Directive, NonEmpty String)]
 directivesFor' str = go directiveStrings'
   where
-  go = map swap . filter ((str `isPrefixOf`) . fst)
+  go = map swap . filter ((str `isPrefixOf`) . NEL.head . fst)
```

Comment:
This change looks like it introduces a bug. Previously, `directivesFor' "h"` would have evaluated to `[(Help, "help")]`. Now, I believe it would be the empty list.

I think `directiveStrings'` should go back to its original type, and `directivesFor'` should remain unchanged. The only intervention needed in `directiveStrings'` should be adding a `toList` in `go`, I think.

Reply:
Great catch! I'm just surprised that there were no tests for this. I fixed it now and added a few basic tests for the `Directive` module.